tests/test_segmentation.py

```diff
@@ -0,0 +1,118 @@
+from nucleus.annotation import SegmentationAnnotation
+from nucleus.dataset import Dataset
+from tests.helpers import (
+    TEST_LOCAL_MASK_URL,
+    TEST_SEGMENTATION_ANNOTATIONS,
+    assert_segmentation_annotation_matches_dict,
+)
+
+
+def test_single_local_semseg_gt_upload(dataset: Dataset):
+    request_annotation = SegmentationAnnotation.from_json(
+        TEST_SEGMENTATION_ANNOTATIONS[0]
+    )
+    request_annotation.mask_url = TEST_LOCAL_MASK_URL
+    response = dataset.annotate(annotations=[request_annotation])
+
+    assert response["dataset_id"] == dataset.id
+    assert response["annotations_processed"] == 1
+    assert response["annotations_ignored"] == 0
+
+    response_annotation = dataset.refloc(request_annotation.reference_id)[
+        "annotations"
+    ]["segmentation"][0]
+
+    assert response_annotation == request_annotation
+
+
+def test_batch_local_semseg_gt_upload(dataset: Dataset):
+
+    # This reference id is not in the dataset.
+    bad_reference_id = TEST_SEGMENTATION_ANNOTATIONS[-1]["reference_id"]
+
+    request_annotations = [
+        SegmentationAnnotation.from_json(json_data)
+        for json_data in TEST_SEGMENTATION_ANNOTATIONS
+    ]
+    for request_annotation in request_annotations:
+        request_annotation.mask_url = TEST_LOCAL_MASK_URL
+    response = dataset.annotate(annotations=request_annotations)
+
+    print(request_annotations)
+    print(response)
+
+    assert response["dataset_id"] == dataset.id
+    assert response["annotations_processed"] == 4
```

Comment:
The upload data structure might well be changed in the future.
```suggestion
    assert response["annotations_processed"] == len(request_annotations)
```